pkg/reconciler/trigger/trigger.go

```diff
@@ -301,3 +365,25 @@ func (r *Reconciler) reconcileSubscription(ctx context.Context, t *v1alpha1.Trig
 	}
 	return newSub, nil
 }
+
+func getObjRefFromDependencyAnnotation(dependencyAnnotation string) (corev1.ObjectReference, error) {
+	var objectRef corev1.ObjectReference
+	if err := json.Unmarshal([]byte(dependencyAnnotation), &objectRef); err != nil {
+		return objectRef, err
+	}
+	return objectRef, nil
+}
+
+func validateDependencyAnnotation(depObjRef corev1.ObjectReference) error {
```

Comment:
Thinking about this more, validating the annotation format might be a job for the webhook instead of the reconciler.

Reply:
Definitely.